Gems/ROS2/Code/Include/ROS2/Sensor/Events/PhysicsBasedSource.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/EBus/OrderedEvent.h>
+#include <AzFramework/Physics/Common/PhysicsTypes.h>
```

Comment:
This was preventing non-unity build for me:
```suggestion
#include <AzFramework/Physics/Common/PhysicsTypes.h>
#include <AzFramework/Physics/PhysicsScene.h>
```